src/query/management/src/role/role_mgr.rs

```diff
@@ -213,13 +195,17 @@ impl RoleMgr {
                 })
                 .await?;
             let mut table_meta = table_info.meta.clone();
+
             // if current owner is not none and not from, return error
             if table_meta.owner.is_some()
                 && table_meta.owner.as_ref().unwrap().owner_role_name != from.clone()
```

Comment:
this privilege check seems missed the checking about the other available roles.

I guess we can extract this privilege check outside here, and put the privilege check logic into the caller side, thus it might easier for us to keep the privilege check logic consistent.

Reply:
caller would loop over all available roles until find one owner.

Reply:
if the caller already have the privilege check, we can remove the check here?

this check here will forbid all the requests whom do not have the exact same roles.